runtime/drivers/mysql/parser.go

```diff
@@ -0,0 +1,307 @@
+package mysql
+
+import (
+	"database/sql"
+	"fmt"
+	"reflect"
+	"strings"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+)
+
+type mapper interface {
+	runtimeType(st reflect.Type) (*runtimev1.Type, error)
+	// dest returns a pointer to a destination value that can be used in Rows.Scan
+	dest(st reflect.Type) (any, error)
+	// value dereferences a pointer created by dest
+	value(p any) (any, error)
+}
```

Comment:
I believe we should still be able to avoid reflection in this code. Based on another common, `dest` might not be needed, and `runtimeType` doesn't need the `st` parameter – the only case where it's currently accessed is in `numericMapper`, and for that case, you can just refactor to this:

```go
type numericMapper struct {
  code runtime.Type_CODE
}

// in getDBTypeNameToMapperMap
m["TINYINT"] = numericMapper{code: runtimev1.Type_CODE_INT8}
...
```

Reply:
Reflection, in this case, doesn't affect performance as it is used only to store scan types (see [mysql driver](https://github.com/go-sql-driver/mysql/blob/master/fields.go#L120)).
`TINYINT` might be scanned as `uint8`, `int8` and `sql.NullInt64` depending on `NOT NULL` and `UNSIGNED` column props